test/e2e/nodepool_autorepair_test.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func testNodePoolAutoRepair(mgmtClient crclient.Client, guestCluster *hyperv1.HostedCluster, guestClient crclient.Client, clusterOpts core.CreateOptions) func(t *testing.T) {
+	return func(t *testing.T) {
+		t.Parallel()
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(context.TODO())
+
+		originalNP := hyperv1.NodePool{}
+		defer func() {
+			t.Log("Context Cancelled in testNodePoolAutoRepair function")
+			cancel()
+		}()
+
+		// List NodePools (should exists only one)
+		nodePools := &hyperv1.NodePoolList{}
+		err := mgmtClient.List(ctx, nodePools, &crclient.ListOptions{
+			Namespace: guestCluster.Namespace,
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepools")
+		for _, nodePool := range nodePools.Items {
+			if !strings.Contains(nodePool.Name, "-test-") {
+				originalNP = nodePool
+			}
+		}
+		g.Expect(originalNP.Name).NotTo(ContainSubstring("test"))
+		awsNPInfo := originalNP.Spec.Platform.AWS
+
+		// Define a new Nodepool
+		nodePool := &hyperv1.NodePool{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "NodePool",
+				APIVersion: hyperv1.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      guestCluster.Name + "-" + "test-autorepair",
+				Namespace: guestCluster.Namespace,
+			},
+			Spec: hyperv1.NodePoolSpec{
+				Management: hyperv1.NodePoolManagement{
+					UpgradeType: hyperv1.UpgradeTypeReplace,
+					AutoRepair:  true,
+				},
+				ClusterName: guestCluster.Name,
+				Replicas:    &oneReplicas,
+				Release: hyperv1.Release{
+					Image: guestCluster.Spec.Release.Image,
+				},
+				Platform: hyperv1.NodePoolPlatform{
+					Type: guestCluster.Spec.Platform.Type,
+					AWS:  awsNPInfo,
+				},
+			},
+		}
+
+		// Create NodePool for current test
+		err = mgmtClient.Create(ctx, nodePool)
+		if err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create nodePool %s with Autorepair function: %v", nodePool.Name, err)
+			}
+
+			// If the NodePool already exists...
+			// This should not happen in CI but it's useful in local tests
+			existantNodePool := &hyperv1.NodePool{}
+			err = mgmtClient.Get(ctx, crclient.ObjectKeyFromObject(nodePool), existantNodePool)
+			g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepool")
+			err = mgmtClient.Delete(ctx, existantNodePool)
+			g.Expect(err).NotTo(HaveOccurred(), "failed to Delete the existant NodePool")
+			t.Logf("waiting for NodePools in-place update with NTO-generated MachineConfig")
```

Comment:
how is this related with autoRepair?

Reply:
This part it's more focused when a nodePool already exists with same name and should be replaced by a new one. This enable us to (in the future) put some kind of "retry" policy in order to avoid false positive issues. The `g.expect..` statements on that block are to ensure (in that case)  are properly deleted and traced into the logs.

Also (not related with he CI) it's quite useful to just relaunch the E2E test and in local check that the nodePool is being recreated without need of manual deletion.

Reply:
I mean
>waiting for NodePools in-place update with NTO-generated MachineConfig

Reply:
cc @jparrill

Reply:
Ohh my bad